File: assignments/haskell/meetup/meetup_test.hs
import Test.HUnit (Assertion, (@=?), runTestTT, Test(..), Counts(..))
import System.Exit (ExitCode(..), exitWith)
import Meetup (Weekday(..), Schedule(..), meetupDay)
import Data.Time.Calendar (fromGregorian)

exitProperly :: IO Counts -> IO ()
exitProperly m = do
  counts <- m
  exitWith $ if failures counts /= 0 || errors counts /= 0 then ExitFailure 1 else ExitSuccess

testCase :: String -> Assertion -> Test
testCase label assertion = TestLabel label (TestCase assertion)

main :: IO ()
main = exitProperly $ runTestTT $ TestList
       [ TestList meetupDayTests ]

-- NOTE: This would be a good opportunity to write some QuickCheck properties.
meetupDayTests :: [Test]
meetupDayTests =
  [ testCase "monteenth of may 2013" $
    fromGregorian 2013 5 13 @=? meetupDay Teenth Monday 2013 5
  , testCase "monteenth of august 2013" $
    fromGregorian 2013 8 19 @=? meetupDay Teenth Monday 2013 8
  , testCase "monteenth of september 2013" $
    fromGregorian 2013 9 16 @=? meetupDay Teenth Monday 2013 9
  , testCase "tuesteenth of march 2013" $
    fromGregorian 2013 3 19 @=? meetupDay Teenth Tuesday 2013 3
  , testCase "tuesteenth of april 2013" $
    fromGregorian 2013 4 16 @=? meetupDay Teenth Tuesday 2013 4
  , testCase "tuesteenth of august 2013" $
    fromGregorian 2013 8 13 @=? meetupDay Teenth Tuesday 2013 8
  , testCase "wednesteenth of january 2013" $
    fromGregorian 2013 1 16 @=? meetupDay Teenth Wednesday 2013 1
  , testCase "wednesteenth of februrary 2013" $
    fromGregorian 2013 2 13 @=? meetupDay Teenth Wednesday 2013 2
  , testCase "wednesteenth of june 2013" $
    fromGregorian 2013 6 19 @=? meetupDay Teenth Wednesday 2013 6
  , testCase "thursteenth of may 2013" $
    fromGregorian 2013 5 16 @=? meetupDay Teenth Thursday 2013 5
  , testCase "thursteenth of june 2013" $
    fromGregorian 2013 6 13 @=? meetupDay Teenth Thursday 2013 6
  , testCase "thursteenth of september 2013" $
    fromGregorian 2013 9 19 @=? meetupDay Teenth Thursday 2013 9
  , testCase "friteenth of april 2013" $
    fromGregorian 2013 4 19 @=? meetupDay Teenth Friday 2013 4
  , testCase "friteenth of august 2013" $
    fromGregorian 2013 8 16@=? meetupDay Teenth Friday 2013 8
  , testCase "friteenth of september 2013" $
    fromGregorian 2013 9 13@=? meetupDay Teenth Friday 2013 9
  , testCase "saturteenth of february 2013" $
    fromGregorian 2013 2 16 @=? meetupDay Teenth Saturday 2013 2
  , testCase "saturteenth of april 2013" $
    fromGregorian 2013 4 13 @=? meetupDay Teenth Saturday 2013 4
  , testCase "saturteenth of october 2013" $
    fromGregorian 2013 10 19 @=? meetupDay Teenth Saturday 2013 10
  , testCase "sunteenth of map 2013" $
    fromGregorian 2013 5 19 @=? meetupDay Teenth Sunday 2013 5
  , testCase "sunteenth of june 2013" $
    fromGregorian 2013 6 16 @=? meetupDay Teenth Sunday 2013 6
  , testCase "sunteenth of october 2013" $
    fromGregorian 2013 10 13 @=? meetupDay Teenth Sunday 2013 10
  , testCase "first Monday of march 2013" $
    fromGregorian 2013 3 4 @=? meetupDay First Monday 2013 3
  , testCase "first Monday of april 2013" $
    fromGregorian 2013 4 1 @=? meetupDay First Monday 2013 4
  , testCase "first Tuesday of may 2013" $
    fromGregorian 2013 5 7 @=? meetupDay First Tuesday 2013 5
  , testCase "first Tuesday of june 2013" $
    fromGregorian 2013 6 4 @=? meetupDay First Tuesday 2013 6
  , testCase "first Wednesday of july 2013" $
    fromGregorian 2013 7 3 @=? meetupDay First Wednesday 2013 7
  , testCase "first Wednesday of august 2013" $
    fromGregorian 2013 8 7 @=? meetupDay First Wednesday 2013 8
  , testCase "first Thursday of september 2013" $
    fromGregorian 2013 9 5 @=? meetupDay First Thursday 2013 9
  , testCase "first Thursday of october 2013" $
    fromGregorian 2013 10 3 @=? meetupDay First Thursday 2013 10
  , testCase "first Friday of november 2013" $
    fromGregorian 2013 11 1 @=? meetupDay First Friday 2013 11
  , testCase "first Friday of december 2013" $
    fromGregorian 2013 12 6 @=? meetupDay First Friday 2013 12
  , testCase "first Saturday of january 2013" $
    fromGregorian 2013 1 5 @=? meetupDay First Saturday 2013 1
  , testCase "first Saturday of february 2013" $
    fromGregorian 2013 2 2 @=? meetupDay First Saturday 2013 2
  , testCase "first Sunday of march 2013" $
    fromGregorian 2013 3 3 @=? meetupDay First Sunday 2013 3
  , testCase "first Sunday of april 2013" $
    fromGregorian 2013 4 7 @=? meetupDay First Sunday 2013 4
  , testCase "second Monday of march 2013" $
    fromGregorian 2013 3 11 @=? meetupDay Second Monday 2013 3
  , testCase "second Monday of april 2013" $
    fromGregorian 2013 4 8 @=? meetupDay Second Monday 2013 4
  , testCase "second Tuesday of may 2013" $
    fromGregorian 2013 5 14 @=? meetupDay Second Tuesday 2013 5
  , testCase "second Tuesday of june 2013" $
    fromGregorian 2013 6 11 @=? meetupDay Second Tuesday 2013 6
  , testCase "second Wednesday of july 2013" $
    fromGregorian 2013 7 10 @=? meetupDay Second Wednesday 2013 7
  , testCase "second Wednesday of august 2013" $
    fromGregorian 2013 8 14 @=? meetupDay Second Wednesday 2013 8
  , testCase "second Thursday of september 2013" $
    fromGregorian 2013 9 12 @=? meetupDay Second Thursday 2013 9
  , testCase "second Thursday of october 2013" $
    fromGregorian 2013 10 10 @=? meetupDay Second Thursday 2013 10
  , testCase "second Friday of november 2013" $
    fromGregorian 2013 11 8 @=? meetupDay Second Friday 2013 11
  , testCase "second Friday of december 2013" $
    fromGregorian 2013 12 13 @=? meetupDay Second Friday 2013 12
  , testCase "second Saturday of january 2013" $
    fromGregorian 2013 1 12 @=? meetupDay Second Saturday 2013 1
  , testCase "second Saturday of february 2013" $
    fromGregorian 2013 2 9 @=? meetupDay Second Saturday 2013 2
  , testCase "second Sunday of march 2013" $
    fromGregorian 2013 3 10 @=? meetupDay Second Sunday 2013 3
  , testCase "second Sunday of april 2013" $
    fromGregorian 2013 4 14 @=? meetupDay Second Sunday 2013 4
  , testCase "third Monday of march 2013" $
    fromGregorian 2013 3 18 @=? meetupDay Third Monday 2013 3
  , testCase "third Monday of april 2013" $
    fromGregorian 2013 4 15 @=? meetupDay Third Monday 2013 4
  , testCase "third Tuesday of may 2013" $
    fromGregorian 2013 5 21 @=? meetupDay Third Tuesday 2013 5
  , testCase "third Tuesday of june 2013" $
    fromGregorian 2013 6 18 @=? meetupDay Third Tuesday 2013 6
  , testCase "third Wednesday of july 2013" $
    fromGregorian 2013 7 17 @=? meetupDay Third Wednesday 2013 7
  , testCase "third Wednesday of august 2013" $
    fromGregorian 2013 8 21 @=? meetupDay Third Wednesday 2013 8
  , testCase "third Thursday of september 2013" $
    fromGregorian 2013 9 19 @=? meetupDay Third Thursday 2013 9
  , testCase "third Thursday of october 2013" $
    fromGregorian 2013 10 17 @=? meetupDay Third Thursday 2013 10
  , testCase "third Friday of november 2013" $
    fromGregorian 2013 11 15 @=? meetupDay Third Friday 2013 11
  , testCase "third Friday of december 2013" $
    fromGregorian 2013 12 20 @=? meetupDay Third Friday 2013 12
  , testCase "third Saturday of january 2013" $
    fromGregorian 2013 1 19 @=? meetupDay Third Saturday 2013 1
  , testCase "third Saturday of february 2013" $
    fromGregorian 2013 2 16 @=? meetupDay Third Saturday 2013 2
  , testCase "third Sunday of march 2013" $
    fromGregorian 2013 3 17 @=? meetupDay Third Sunday 2013 3
  , testCase "third Sunday of april 2013" $
    fromGregorian 2013 4 21 @=? meetupDay Third Sunday 2013 4
  , testCase "fourth Monday of march 2013" $
    fromGregorian 2013 3 25 @=? meetupDay Fourth Monday 2013 3
  , testCase "fourth Monday of april 2013" $
    fromGregorian 2013 4 22 @=? meetupDay Fourth Monday 2013 4
  , testCase "fourth Tuesday of may 2013" $
    fromGregorian 2013 5 28 @=? meetupDay Fourth Tuesday 2013 5
  , testCase "fourth Tuesday of june 2013" $
    fromGregorian 2013 6 25 @=? meetupDay Fourth Tuesday 2013 6
  , testCase "fourth Wednesday of july 2013" $
    fromGregorian 2013 7 24 @=? meetupDay Fourth Wednesday 2013 7
  , testCase "fourth Wednesday of august 2013" $
    fromGregorian 2013 8 28 @=? meetupDay Fourth Wednesday 2013 8
  , testCase "fourth Thursday of september 2013" $
    fromGregorian 2013 9 26 @=? meetupDay Fourth Thursday 2013 9
  , testCase "fourth Thursday of october 2013" $
    fromGregorian 2013 10 24 @=? meetupDay Fourth Thursday 2013 10
  , testCase "fourth Friday of november 2013" $
    fromGregorian 2013 11 22 @=? meetupDay Fourth Friday 2013 11
  , testCase "fourth Friday of december 2013" $
    fromGregorian 2013 12 27 @=? meetupDay Fourth Friday 2013 12
  , testCase "fourth Saturday of january 2013" $
    fromGregorian 2013 1 26 @=? meetupDay Fourth Saturday 2013 1
  , testCase "fourth Saturday of february 2013" $
    fromGregorian 2013 2 23 @=? meetupDay Fourth Saturday 2013 2
  , testCase "fourth Sunday of march 2013" $
    fromGregorian 2013 3 24 @=? meetupDay Fourth Sunday 2013 3
  , testCase "fourth Sunday of april 2013" $
    fromGregorian 2013 4 28 @=? meetupDay Fourth Sunday 2013 4
  , testCase "last Monday of march 2013" $
    fromGregorian 2013 3 25 @=? meetupDay Last Monday 2013 3
  , testCase "last Monday of april 2013" $
    fromGregorian 2013 4 29 @=? meetupDay Last Monday 2013 4
  , testCase "last Tuesday of may 2013" $
    fromGregorian 2013 5 28 @=? meetupDay Last Tuesday 2013 5
  , testCase "last Tuesday of june 2013" $
    fromGregorian 2013 6 25 @=? meetupDay Last Tuesday 2013 6
  , testCase "last Wednesday of july 2013" $
    fromGregorian 2013 7 31 @=? meetupDay Last Wednesday 2013 7
  , testCase "last Wednesday of august 2013" $
    fromGregorian 2013 8 28 @=? meetupDay Last Wednesday 2013 8
  , testCase "last Thursday of september 2013" $
    fromGregorian 2013 9 26 @=? meetupDay Last Thursday 2013 9
  , testCase "last Thursday of october 2013" $
    fromGregorian 2013 10 31 @=? meetupDay Last Thursday 2013 10
  , testCase "last Friday of november 2013" $
    fromGregorian 2013 11 29 @=? meetupDay Last Friday 2013 11
  , testCase "last Friday of december 2013" $
    fromGregorian 2013 12 27 @=? meetupDay Last Friday 2013 12
  , testCase "last Saturday of january 2013" $
    fromGregorian 2013 1 26 @=? meetupDay Last Saturday 2013 1
  , testCase "last Saturday of february 2013" $
    fromGregorian 2013 2 23 @=? meetupDay Last Saturday 2013 2
  , testCase "last Sunday of march 2013" $
    fromGregorian 2013 3 31 @=? meetupDay Last Sunday 2013 3
  , testCase "last Sunday of april 2013" $
    fromGregorian 2013 4 28 @=? meetupDay Last Sunday 2013 4
  ]

File: assignments/scala/meetup/src/test/scala/meetup_test.scala
import org.scalatest._
import java.util.GregorianCalendar

class MeetupTest extends FunSuite with Matchers {
  //Note: Java uses 0-indexed months for GregorianCalendar
  test ("monteenth of may 2013") {
    Meetup(5, 2013).teenth(Meetup.Mon) should be (new GregorianCalendar(2013, 4, 13))
  }

  test ("monteenth of august 2013") {
    pending
    Meetup(8, 2013).teenth(Meetup.Mon) should be (new GregorianCalendar(2013, 7, 19))
  }

  test ("monteenth of september 2013") {
    pending
    Meetup(9, 2013).teenth(Meetup.Mon) should be (new GregorianCalendar(2013, 8, 16))
  }

  test ("tuesteenth of march 2013") {
    pending
    Meetup(3, 2013).teenth(Meetup.Tue) should be (new GregorianCalendar(2013, 2, 19))
  }

  test ("tuesteenth of april 2013") {
    pending
    Meetup(4, 2013).teenth(Meetup.Tue) should be (new GregorianCalendar(2013, 3, 16))
  }

  test ("tuesteenth of august 2013") {
    pending
    Meetup(8, 2013).teenth(Meetup.Tue) should be (new GregorianCalendar(2013, 7, 13))
  }

  test ("wednesteenth of january 2013") {
    pending
    Meetup(1, 2013).teenth(Meetup.Wed) should be (new GregorianCalendar(2013, 0, 16))
  }

  test ("wednesteenth of february 2013") {
    pending
    Meetup(2, 2013).teenth(Meetup.Wed) should be (new GregorianCalendar(2013, 1, 13))
  }

  test ("wednesteenth of june 2013") {
    pending
    Meetup(6, 2013).teenth(Meetup.Wed) should be (new GregorianCalendar(2013, 5, 19))
  }

  test ("thursteenth of may 2013") {
    pending
    Meetup(5, 2013).teenth(Meetup.Thu) should be (new GregorianCalendar(2013, 4, 16))
  }

  test ("thursteenth of june 2013") {
    pending
    Meetup(6, 2013).teenth(Meetup.Thu) should be (new GregorianCalendar(2013, 5, 13))
  }

  test ("thursteenth of september 2013") {
    pending
    Meetup(9, 2013).teenth(Meetup.Thu) should be (new GregorianCalendar(2013, 8, 19))
  }

  test ("friteenth of april 2013") {
    pending
    Meetup(4, 2013).teenth(Meetup.Fri) should be (new GregorianCalendar(2013, 3, 19))
  }

  test ("friteenth of august 2013") {
    pending
    Meetup(8, 2013).teenth(Meetup.Fri) should be (new GregorianCalendar(2013, 7, 16))
  }

  test ("friteenth of september 2013") {
    pending
    Meetup(9, 2013).teenth(Meetup.Fri) should be (new GregorianCalendar(2013, 8, 13))
  }

  test ("saturteenth of february 2013") {
    pending
    Meetup(2, 2013).teenth(Meetup.Sat) should be (new GregorianCalendar(2013, 1, 16))
  }

  test ("saturteenth of april 2013") {
    pending
    Meetup(4, 2013).teenth(Meetup.Sat) should be (new GregorianCalendar(2013, 3, 13))
  }

  test ("saturteenth of october 2013") {
    pending
    Meetup(10, 2013).teenth(Meetup.Sat) should be (new GregorianCalendar(2013, 9, 19))
  }

  test ("sunteenth of map 2013") {
    pending
    Meetup(5, 2013).teenth(Meetup.Sun) should be (new GregorianCalendar(2013, 4, 19))
  }

  test ("sunteenth of june 2013") {
    pending
    Meetup(6, 2013).teenth(Meetup.Sun) should be (new GregorianCalendar(2013, 5, 16))
  }

  test ("sunteenth of october 2013") {
    pending
    Meetup(10, 2013).teenth(Meetup.Sun) should be (new GregorianCalendar(2013, 9, 13))
  }

  test ("first monday of march 2013") {
    pending
    Meetup(3, 2013).first(Meetup.Mon) should be (new GregorianCalendar(2013, 2, 4))
  }

  test ("first monday of april 2013") {
    pending
    Meetup(4, 2013).first(Meetup.Mon) should be (new GregorianCalendar(2013, 3, 1))
  }

  test ("first tuesday of may 2013") {
    pending
    Meetup(5, 2013).first(Meetup.Tue) should be (new GregorianCalendar(2013, 4, 7))
  }

  test ("first tuesday of june 2013") {
    pending
    Meetup(6, 2013).first(Meetup.Tue) should be (new GregorianCalendar(2013, 5, 4))
  }

  test ("first wednesday of july 2013") {
    pending
    Meetup(7, 2013).first(Meetup.Wed) should be (new GregorianCalendar(2013, 6, 3))
  }

  test ("first wednesday of august 2013") {
    pending
    Meetup(8, 2013).first(Meetup.Wed) should be (new GregorianCalendar(2013, 7, 7))
  }

  test ("first thursday of september 2013") {
    pending
    Meetup(9, 2013).first(Meetup.Thu) should be (new GregorianCalendar(2013, 8, 5))
  }

  test ("first thursday of october 2013") {
    pending
    Meetup(10, 2013).first(Meetup.Thu) should be (new GregorianCalendar(2013, 9, 3))
  }

  test ("first friday of november 2013") {
    pending
    Meetup(11, 2013).first(Meetup.Fri) should be (new GregorianCalendar(2013, 10, 1))
  }

  test ("first friday of december 2013") {
    pending
    Meetup(12, 2013).first(Meetup.Fri) should be (new GregorianCalendar(2013, 11, 6))
  }

  test ("first saturday of january 2013") {
    pending
    Meetup(1, 2013).first(Meetup.Sat) should be (new GregorianCalendar(2013, 0, 5))
  }

  test ("first saturday of february 2013") {
    pending
    Meetup(2, 2013).first(Meetup.Sat) should be (new GregorianCalendar(2013, 1, 2))
  }

  test ("first sunday of march 2013") {
    pending
    Meetup(3, 2013).first(Meetup.Sun) should be (new GregorianCalendar(2013, 2, 3))
  }

  test ("first sunday of april 2013") {
    pending
    Meetup(4, 2013).first(Meetup.Sun) should be (new GregorianCalendar(2013, 3, 7))
  }

  test ("second monday of march 2013") {
    pending
    Meetup(3, 2013).second(Meetup.Mon) should be (new GregorianCalendar(2013, 2, 11))
  }

  test ("second monday of april 2013") {
    pending
    Meetup(4, 2013).second(Meetup.Mon) should be (new GregorianCalendar(2013, 3, 8))
  }

  test ("second tuesday of may 2013") {
    pending
    Meetup(5, 2013).second(Meetup.Tue) should be (new GregorianCalendar(2013, 4, 14))
  }

  test ("second tuesday of june 2013") {
    pending
    Meetup(6, 2013).second(Meetup.Tue) should be (new GregorianCalendar(2013, 5, 11))
  }

  test ("second wednesday of july 2013") {
    pending
    Meetup(7, 2013).second(Meetup.Wed) should be (new GregorianCalendar(2013, 6, 10))
  }

  test ("second wednesday of august 2013") {
    pending
    Meetup(8, 2013).second(Meetup.Wed) should be (new GregorianCalendar(2013, 7, 14))
  }

  test ("second thursday of september 2013") {
    pending
    Meetup(9, 2013).second(Meetup.Thu) should be (new GregorianCalendar(2013, 8, 12))
  }

  test ("second thursday of october 2013") {
    pending
    Meetup(10, 2013).second(Meetup.Thu) should be (new GregorianCalendar(2013, 9, 10))
  }

  test ("second friday of november 2013") {
    pending
    Meetup(11, 2013).second(Meetup.Fri) should be (new GregorianCalendar(2013, 10, 8))
  }

  test ("second friday of december 2013") {
    pending
    Meetup(12, 2013).second(Meetup.Fri) should be (new GregorianCalendar(2013, 11, 13))
  }

  test ("second saturday of january 2013") {
    pending
    Meetup(1, 2013).second(Meetup.Sat) should be (new GregorianCalendar(2013, 0, 12))
  }

  test ("second saturday of february 2013") {
    pending
    Meetup(2, 2013).second(Meetup.Sat) should be (new GregorianCalendar(2013, 1, 9))
  }

  test ("second sunday of march 2013") {
    pending
    Meetup(3, 2013).second(Meetup.Sun) should be (new GregorianCalendar(2013, 2, 10))
  }

  test ("second sunday of april 2013") {
    pending
    Meetup(4, 2013).second(Meetup.Sun) should be (new GregorianCalendar(2013, 3, 14))
  }

  test ("third monday of march 2013") {
    pending
    Meetup(3, 2013).third(Meetup.Mon) should be (new GregorianCalendar(2013, 2, 18))
  }

  test ("third monday of april 2013") {
    pending
    Meetup(4, 2013).third(Meetup.Mon) should be (new GregorianCalendar(2013, 3, 15))
  }

  test ("third tuesday of may 2013") {
    pending
    Meetup(5, 2013).third(Meetup.Tue) should be (new GregorianCalendar(2013, 4, 21))
  }

  test ("third tuesday of june 2013") {
    pending
    Meetup(6, 2013).third(Meetup.Tue) should be (new GregorianCalendar(2013, 5, 18))
  }

  test ("third wednesday of july 2013") {
    pending
    Meetup(7, 2013).third(Meetup.Wed) should be (new GregorianCalendar(2013, 6, 17))
  }

  test ("third wednesday of august 2013") {
    pending
    Meetup(8, 2013).third(Meetup.Wed) should be (new GregorianCalendar(2013, 7, 21))
  }

  test ("third thursday of september 2013") {
    pending
    Meetup(9, 2013).third(Meetup.Thu) should be (new GregorianCalendar(2013, 8, 19))
  }

  test ("third thursday of october 2013") {
    pending
    Meetup(10, 2013).third(Meetup.Thu) should be (new GregorianCalendar(2013, 9, 17))
  }

  test ("third friday of november 2013") {
    pending
    Meetup(11, 2013).third(Meetup.Fri) should be (new GregorianCalendar(2013, 10, 15))
  }

  test ("third friday of december 2013") {
    pending
    Meetup(12, 2013).third(Meetup.Fri) should be (new GregorianCalendar(2013, 11, 20))
  }

  test ("third saturday of january 2013") {
    pending
    Meetup(1, 2013).third(Meetup.Sat) should be (new GregorianCalendar(2013, 0, 19))
  }

  test ("third saturday of february 2013") {
    pending
    Meetup(2, 2013).third(Meetup.Sat) should be (new GregorianCalendar(2013, 1, 16))
  }

  test ("third sunday of march 2013") {
    pending
    Meetup(3, 2013).third(Meetup.Sun) should be (new GregorianCalendar(2013, 2, 17))
  }

  test ("third sunday of april 2013") {
    pending
    Meetup(4, 2013).third(Meetup.Sun) should be (new GregorianCalendar(2013, 3, 21))
  }

  test ("fourth monday of march 2013") {
    pending
    Meetup(3, 2013).fourth(Meetup.Mon) should be (new GregorianCalendar(2013, 2, 25))
  }

  test ("fourth monday of april 2013") {
    pending
    Meetup(4, 2013).fourth(Meetup.Mon) should be (new GregorianCalendar(2013, 3, 22))
  }

  test ("fourth tuesday of may 2013") {
    pending
    Meetup(5, 2013).fourth(Meetup.Tue) should be (new GregorianCalendar(2013, 4, 28))
  }

  test ("fourth tuesday of june 2013") {
    pending
    Meetup(6, 2013).fourth(Meetup.Tue) should be (new GregorianCalendar(2013, 5, 25))
  }

  test ("fourth wednesday of july 2013") {
    pending
    Meetup(7, 2013).fourth(Meetup.Wed) should be (new GregorianCalendar(2013, 6, 24))
  }

  test ("fourth wednesday of august 2013") {
    pending
    Meetup(8, 2013).fourth(Meetup.Wed) should be (new GregorianCalendar(2013, 7, 28))
  }

  test ("fourth thursday of september 2013") {
    pending
    Meetup(9, 2013).fourth(Meetup.Thu) should be (new GregorianCalendar(2013, 8, 26))
  }

  test ("fourth thursday of october 2013") {
    pending
    Meetup(10, 2013).fourth(Meetup.Thu) should be (new GregorianCalendar(2013, 9, 24))
  }

  test ("fourth friday of november 2013") {
    pending
    Meetup(11, 2013).fourth(Meetup.Fri) should be (new GregorianCalendar(2013, 10, 22))
  }

  test ("fourth friday of december 2013") {
    pending
    Meetup(12, 2013).fourth(Meetup.Fri) should be (new GregorianCalendar(2013, 11, 27))
  }

  test ("fourth saturday of january 2013") {
    pending
    Meetup(1, 2013).fourth(Meetup.Sat) should be (new GregorianCalendar(2013, 0, 26))
  }

  test ("fourth saturday of february 2013") {
    pending
    Meetup(2, 2013).fourth(Meetup.Sat) should be (new GregorianCalendar(2013, 1, 23))
  }

  test ("fourth sunday of march 2013") {
    pending
    Meetup(3, 2013).fourth(Meetup.Sun) should be (new GregorianCalendar(2013, 2, 24))
  }

  test ("fourth sunday of april 2013") {
    pending
    Meetup(4, 2013).fourth(Meetup.Sun) should be (new GregorianCalendar(2013, 3, 28))
  }

  test ("last monday of march 2013") {
    pending
    Meetup(3, 2013).last(Meetup.Mon) should be (new GregorianCalendar(2013, 2, 25))
  }

  test ("last monday of april 2013") {
    pending
    Meetup(4, 2013).last(Meetup.Mon) should be (new GregorianCalendar(2013, 3, 29))
  }

  test ("last tuesday of may 2013") {
    pending
    Meetup(5, 2013).last(Meetup.Tue) should be (new GregorianCalendar(2013, 4, 28))
  }

  test ("last tuesday of june 2013") {
    pending
    Meetup(6, 2013).last(Meetup.Tue) should be (new GregorianCalendar(2013, 5, 25))
  }

  test ("last wednesday of july 2013") {
    pending
    Meetup(7, 2013).last(Meetup.Wed) should be (new GregorianCalendar(2013, 6, 31))
  }

  test ("last wednesday of august 2013") {
    pending
    Meetup(8, 2013).last(Meetup.Wed) should be (new GregorianCalendar(2013, 7, 28))
  }

  test ("last thursday of september 2013") {
    pending
    Meetup(9, 2013).last(Meetup.Thu) should be (new GregorianCalendar(2013, 8, 26))
  }

  test ("last thursday of october 2013") {
    pending
    Meetup(10, 2013).last(Meetup.Thu) should be (new GregorianCalendar(2013, 9, 31))
  }

  test ("last friday of november 2013") {
    pending
    Meetup(11, 2013).last(Meetup.Fri) should be (new GregorianCalendar(2013, 10, 29))
  }

  test ("last friday of december 2013") {
    pending
    Meetup(12, 2013).last(Meetup.Fri) should be (new GregorianCalendar(2013, 11, 27))
  }

  test ("last saturday of january 2013") {
    pending
    Meetup(1, 2013).last(Meetup.Sat) should be (new GregorianCalendar(2013, 0, 26))
  }

  test ("last saturday of february 2013") {
    pending
    Meetup(2, 2013).last(Meetup.Sat) should be (new GregorianCalendar(2013, 1, 23))
  }

  test ("last sunday of march 2013") {
    pending
    Meetup(3, 2013).last(Meetup.Sun) should be (new GregorianCalendar(2013, 2, 31))
  }

  test ("last sunday of april 2013") {
    pending
    Meetup(4, 2013).last(Meetup.Sun) should be (new GregorianCalendar(2013, 3, 28))
  }
}
